keps/sig-node/20190226-pod-overhead.md

```diff
@@ -93,157 +117,119 @@ so should not be impacted by pod Overhead.
   }
 }
-```
 
-For scheduling, the pod resource requests are added to the container resource requests.
+For scheduling, the pod `Overhead` resource requests are added to the container resource requests.
 
 We don't currently enforce resource limits on the pod cgroup, but this becomes feasible once
 pod overhead is accountable. If the pod specifies a resource limit, and all containers in the
 pod specify a limit, then the sum of those limits becomes a pod-level limit, enforced through the
 pod cgroup.
 
-Users are not expected to manually set the pod resources; if a runtimeClass is being utilized,
-the manual value will be discarded. See RuntimeController for the proposal for setting these
-resources.
+Users are not expected to manually set `Overhead`; any prior values will being set will result in the workload
+being rejected. If runtimeClass is configured and selected in the PodSpec, `Overhead` will be set to the value
+defined in the corresponding runtimeClass. This is described in detail in
+[RuntimeClass admission controller](#runtimeclass-admission-controller).
+
+Being able to specify resource requirements for a workload at the pod level instead of container
+level has been discussed, but isn't proposed in this KEP.
+
+In the event that pod-level requirements are introduced, pod overhead should be kept separate. This
+simplifies several scenarios:
+ - overhead, once added to the spec, stays with the workload, even if runtimeClass is redefined
+ or removed.
+ - the pod spec can be referenced directly from scheduler, resourceQuote controller and kubelet,
+ instead of referencing a runtimeClass object which could have possibly been removed.
 
-Being able to specify resource requirements for a workload at the pod level instead of container level
-has been discussed, but isn't proposed in this KEP.
+#### Container Runtime Interface (CRI)
 
-In the event that pod-level requirements are introduced, pod overhead should be kept separate. This simplifies
-several scenarios:
- - overhead, once added to the spec, stays with the workload, even if runtimeClass is redefined or removed.
- - the pod spec can be referenced directly from scheduler, resourceQuote controller and kubelet, instead of referencing
- a runtimeClass object which could have possibly been removed.
+The pod cgroup is managed by the Kubelet, so passing the pod-level resource to the CRI implementation
+is not strictly necessary. However, some runtimes may wish to take advantage of this information, for
+instance for sizing the Kata Container VM.
+
+LinuxContainerResources is added to the LinuxPodSandboxConfig for both overhead and container
+totals, as optional fields:
+
+type LinuxPodSandboxConfig struct {
+	Overhead *LinuxContainerResources
+	ContainerResources *LinuxContainerResources
+}
 
+Unlike the Resources field on the Kubernetes PodSpec, the ContainerResources field
+in the LinuxPodSandboxConfig matches the pod-level limits (i.e. total of container limits).
```

Comment:
Can you clarify what this sentence means. It is not quite clear.

Reply:
What happens when container resource update path?

Reply:
As of today excluding VPA no container resource gets updated on the fly except cpusets. Hence having this allows us to size the VM upfront. And cpusets will in any case be applied purely on the host at a POD level. So it is good to have this at pod level allowing the VM lifecycle to accommodate simpler hypervisors which do not support cpu and memory hotplug.